### PR TITLE
Upgrade `2.x` compiler baseline to Java 17

### PR DESCRIPTION
**Pull request by vy**
*Warning!* `revision` needs to be adjusted prior to merge.

### CODE REVIEW

File: .github/dependabot.yaml [hunk 1]
```diff
@@ -43,21 +43,12 @@ updates:
     # Keep Logback version 1.2.x
     - dependency-name: "ch.qos.logback:*"
       update-types: ["version-update:semver-major", "version-update:semver-minor"]
-    # Mockito 5.x requires Java 11
-    - dependency-name: "org.mockito:*"
-      update-types: ["version-update:semver-major"]
-    # JUnit Pioneer 2.x requires Java 11
-    - dependency-name: "org.junit-pioneer:*"
-      update-types: ["version-update:semver-major"]
```
Comment (ppkarwasz) [inline .github/dependabot.yaml:51]: Are you sure, we can upgrade these?

These are test dependencies and tests are still compiled with `-release 8` and **must** run on JRE 8.

File: .github/dependabot.yaml [hunk 2]
```diff
@@ -71,15 +62,9 @@ updates:
       update-types: ["version-update:semver-minor", "version-update:semver-patch"]
     - dependency-name: "org.openrewrite.recipe:*"
       update-types: ["version-update:semver-minor", "version-update:semver-patch"]
-    # Json Unit 3.x requires Java 17
-    - dependency-name: "net.javacrumbs.json-unit:*"
-      versions: ["[3.0.0,)"]
```
Comment (ppkarwasz) [inline .github/dependabot.yaml:76]: Same as above.

File: .github/dependabot.yaml [hunk 3]
```diff
@@ -71,15 +62,9 @@ updates:
...
     # LMAX Disruptor requires Java 11
     - dependency-name: "com.lmax:disruptor"
       versions: ["[4.0.0,)"]
-    # WebCompere System Stubs requires Java 11
-    - dependency-name: "uk.org.webcompere:*"
-      versions: ["2.1.0,)"]
```
Comment (ppkarwasz) [inline .github/dependabot.yaml:82]: Also must be `< 2.1.0`.

**Review by ppkarwasz — CHANGES_REQUESTED** [review]
@vy,

Great job.

Some deps must however be downgraded, since tests must be runnable also on Java 8.

File: .github/dependabot.yaml (hunk 1 above)
Comment (vy) [reply .github/dependabot.yaml]: :+1: Reverted.

File: .github/dependabot.yaml (hunk 2 above)
Comment (vy) [reply .github/dependabot.yaml]: :+1: Reverted.

File: .github/dependabot.yaml (hunk 3 above)
Comment (vy) [reply .github/dependabot.yaml]: :+1: Reverted.

**Review by ppkarwasz — APPROVED** [review]
LGTM

**Comment by vy** [comment]
> Some deps must however be downgraded, since tests must be runnable also on Java 8.

Thanks so much for the review, much appreciated! :pray: I have reverted test dependency upgrades. I will merge the changes sometime today.

**Pull request merged**